docs/source/example_notebooks/functional_api.ipynb

```diff
@@ -0,0 +1,219 @@
+    "\n",
+    "data = dowhy.datasets.linear_dataset(\n",
+    "    beta=4,\n",
+    "    num_common_causes=2,\n",
+    "    num_instruments=1,\n",
+    "    num_effect_modifiers=1,\n",
+    "    num_samples=500,\n",
+    "    treatment_is_binary=True,\n",
+    "    stddev_treatment_noise=10,\n",
+    "    num_discrete_common_causes=1,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "treatment_name = data[\"treatment_name\"]\n",
+    "outcome_name = data[\"outcome_name\"]\n",
+    "\n",
+    "graph = CausalGraph(\n",
+    "    treatment_name=treatment_name,\n",
+    "    outcome_name=outcome_name,\n",
+    "    graph=data[\"gml_graph\"],\n",
+    "    effect_modifier_names=data[\"effect_modifier_names\"],\n",
+    "    common_cause_names=data[\"common_causes_names\"],\n",
+    "    observed_node_names=data[\"df\"].columns.tolist(),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Functional API Usage"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# New functional API (AutoIdentifier)\n",
+    "\n",
+    "identified_estimand_auto = dw.identify_effect(\n",
+    "    graph=graph,\n",
+    "    treatment=treatment_name,\n",
+    "    outcome=outcome_name,\n",
+    "    method=AutoIdentifier(estimand_type=dw.EstimandType.NONPARAMETRIC_ATE),\n",
+    ")\n",
+    "print(identified_estimand_auto)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# New functional API (IDIdentifier)\n",
+    "identified_estimand_id = dw.identify_effect(\n",
+    "    graph=graph,\n",
+    "    treatment=treatment_name,\n",
+    "    outcome=outcome_name,\n",
+    "    method=IDIdentifier(estimand_type=dw.EstimandType.NONPARAMETRIC_ATE),\n",
+    ")\n",
+    "print(identified_estimand_id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Backwards compatibility\n",
+    "\n",
+    "This section shows the same results as the new Functional API with the old `CausalModel` API"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# CausalModel API with AutoIdentifier\n",
+    "\n",
+    "model = dw.CausalModel(\n",
+    "    data=data[\"df\"],\n",
+    "    treatment=data[\"treatment_name\"],\n",
+    "    outcome=data[\"outcome_name\"],\n",
+    "    graph=data[\"gml_graph\"],\n",
+    ")\n",
+    "\n",
+    "identified_estimand = model.identify_effect()\n",
+    "print(identified_estimand)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Another way of executing the identify effect by directly calling the object identify_effect method\n",
+    "\n",
```

Comment:
This alternative way is a part of the new API, so I suggest moving it outside of backwards compatibility. Backwards compatibility should only include the code with CausalModel class.There can be another markdown section that comes before the backwards compatibility section, titled "For advanced users: Directly accessing the identifier object".